extensions/usage-metrics-dashboard/app.R

```diff
@@ -274,25 +275,80 @@ ui <- function(request) {
 server <- function(input, output, session) {
   # Set up Connect client; handle error if Visitor API Key integration isn't
   # present.
+
+  publisher_client <- connect()
+
+  selected_integration_guid <- reactiveVal(NULL)
+  observeEvent(input$auto_add_integration, {
+    auto_add_integration(publisher_client, selected_integration_guid())
+    # Hard refresh so that the sidebar gets the up to date info
+    runjs("window.top.location.reload(true);")
+  })
+
   client <- NULL
   tryCatch(
     client <- connect(
       token = session$request$HTTP_POSIT_CONNECT_USER_SESSION_TOKEN
     ),
     error = function(e) {
-      showModal(modalDialog(
-        title = "Additional Setup Required",
-        footer = NULL,
-        HTML(paste(
-          "In the Access panel to the right, click <strong>\"Add integration\"</strong>,",
-          "then select a <strong>Visitor API Key</strong> integration.",
-          "If you don't see one in the list, an administrator must enable this feature on your Connect server.",
-          "See the <a href='https://docs.posit.co/connect/admin/integrations/oauth-integrations/connect/' target='_blank'>Admin Guide</a> for setup instructions.",
+      eligible_integrations <- get_eligible_integrations(publisher_client)
+      selected_integration <- slice_head(eligible_integrations, n = 1)
+      selected_integration_guid(selected_integration$guid)
+
+      if (nrow(selected_integration) == 1) {
+        message <- paste0(
+          "This content uses a <strong>Visitor API Key</strong> ",
+          "integration to show users their own content's usage data.",
           "<br><br>",
-          "For guidance on using visitor-scoped permissions in your own Connect apps, see the",
-          "<a href='https://docs.posit.co/connect/user/oauth-integrations/#obtaining-a-visitor-api-key' target='_blank'>User Guide</a>.",
-          sep = " "
+          "Click the button below to allow this content to use the integration <strong>'",
+          selected_integration$name,
+          "'</strong>.",
+          "<br><br>",
+          "For more information, see ",
+          "<a href='https://docs.posit.co/connect/user/oauth-integrations/#obtaining-a-visitor-api-key' ",
+          "target='_blank'>Visitor API Key section of the User Guide</a>."
+        )
```

Comment:
Lets change this to something like this, where the name of the integration is in the button and we mention that it's compatible rather than details about it.

> This content requires a Visitor API Key integration to show users the content they have access to. A compatible integration is listed below.
> 
> For more information, see documentation on Visitor API Key integrations.
> 
> [+ Add the "Connect API - Publisher role" integration]